soroban-env-host/src/budget.rs

```diff
@@ -779,7 +478,22 @@ impl Debug for BudgetImpl {
             )?;
         }
         writeln!(f, "{:=<165}", "")?;
+        writeln!(
+            f,
+            "Internal details (diagnostics info, does not affect fees) "
```

Comment:
I'm not sure how I feel about the name 'internal'. I recall we've discussed 'debug' budget. That's not the best term either, but I think we could brainstorm this more, maybe something like 'diagnostics_budget'?
I also noticed that you call this 'debug_budget' in host, so it's even more confusing this way.

Reply:
Yeah it is definitely debatable. The choice of "internal" was a result of several considerations.
The internal budget is intended to cover two broad categories of work:
1. Debug, diagnostic work. These work happen in the production (non-test, non-preflight) workflow, but must have `is_debug` turned on. This is previously the "with_free_budget" use case for all diagnostic events and logging-related.
2. Testing, preflight. Most of these are intended for preserving production budget number, such as in the recording auth workflow.  But it may also include any other code that we just "don't care" about budgeting, such as for testing. This can only be activated for non-production scenarios, and the host can have `is_debug` either `true` or `false`.

Given these considerations, it appears like "internal" is the most accurate term from `budget`'s perspective. It is just an internal counter + limit that is not exposed to the user, and the budget doesn't care which scenario or workflow it came from. 

Note the host has a wrapping version called `with_debug_budget`, which asserts `is_debug()` is `true`. From the host perspective, this interface is the entry point for category 1 -- for all debug, diagnostic work where the `DEBUG` mode must be turned on.

Reply:
Yeah, I discussed this with @jayz22 on friday and we agreed that given the intended dual-use in both diagnostics and preflight, something more-general than `debug` is appropriate. Other terms I've considered here: "shadow budget", "ephemeral budget", "secret budget", "fallback budget", "unofficial budget". @dmkozh do you have any preference?

Reply:
Yeah, I discussed this with @jayz22 on friday and we agreed that given the intended dual-use in both diagnostics and preflight, something more-general than `debug` is appropriate. Other terms I've considered here: "shadow budget", "ephemeral budget", "secret budget", "fallback budget", "unofficial budget". @dmkozh do you have any preference?

Reply:
Ok, that makes sense. I'm still not 100% sure if there is a better name, but 'internal' seems good enough for now.

Reply:
'Shadow budget' might be fine as well. While the budget itself is conceptually unobservable, its name is very much observable by the user (IIUC whoever prints out budget will see it), so we need something not too confusing.

Reply:
I renamed all the "internal budget" to "shadow budget". I agree the word "internal" might be a bit overloaded and could cause confusion.